sccp/imports.py

```diff
@@ -7,19 +7,27 @@
 from sklearn.utils.sparsefuncs import inplace_column_scale, mean_variance_axis
 
 
-def prepare_dataset(X: anndata.AnnData) -> anndata.AnnData:
+def prepare_dataset(X: anndata.AnnData, condition_name: str) -> anndata.AnnData:
     assert isinstance(X.X, spmatrix)
     assert np.amin(X.X.data) >= 0.0  # type: ignore
 
     readSum, _ = mean_variance_axis(X.X, axis=0)  # type: ignore
-    readSum *= X.X.shape[0]
-    inplace_column_scale(X.X, 1.0 / readSum)
+    readSum *= X.shape[0]
+    inplace_column_scale(X.X, 1.0 / (readSum + 1.0e-9))
 
     # Transform values
     X.X.data = np.log10((1000.0 * X.X.data) + 1.0)  # type: ignore
 
+    # Get the indices for subsetting the data
+    _, sgIndex = np.unique(X.obs_vector(condition_name), return_inverse=True)
+    X.obs["condition_unique_idxs"] = sgIndex
+
+    # Pre-calculate gene means
+    means, _ = mean_variance_axis(X.X, axis=0)  # type: ignore
+    X.var["means"] = means
+
     # Filter out genes with too few reads
-    X = X[:, readSum > 1000]
+    X = X[:, readSum > 100]
```

Comment:
Lowered the threshold for including a gene. This includes many more genes within the Thomson study.

Reply:
Yes. Previously we were removing more.